src/client/service/identitiesInfoGet.ts

```diff
@@ -18,40 +22,90 @@ function identitiesInfoGet({
   authenticate: Authenticate;
 }) {
   return async (
-    call: grpc.ServerUnaryCall<identitiesPB.Provider, identitiesPB.Provider>,
-    callback: grpc.sendUnaryData<identitiesPB.Provider>,
+    call: grpc.ServerWritableStream<
+      identitiesPB.ProviderSearch,
+      identitiesPB.Info
+    >,
   ): Promise<void> => {
+    const genWritable = grpcUtils.generatorWritable(call);
     try {
-      const response = new identitiesPB.Provider();
       const metadata = await authenticate(call.metadata);
       call.sendMetadata(metadata);
       const {
-        providerId,
+        providerIds,
+        identityId,
       }: {
-        providerId: ProviderId;
+        providerIds: Array<ProviderId>;
+        identityId: IdentityId;
       } = validateSync(
         (keyPath, value) => {
           return matchSync(keyPath)(
-            [['providerId'], () => validationUtils.parseProviderId(value)],
+            [['providerIds'], () => value.map(validationUtils.parseProviderId)],
+            [['identityId'], () => validationUtils.parseIdentityId(value)],
             () => value,
           );
         },
         {
-          providerId: call.request.getProviderId(),
+          providerIds: call.request.getProviderIdList(),
+          identityId: call.request.getIdentityId(),
         },
       );
-      const provider = identitiesManager.getProvider(providerId);
-      if (provider !== undefined) {
-        const identities = await provider.getAuthIdentityIds();
-        response.setProviderId(providerId);
-        if (identities.length !== 0) {
-          response.setIdentityId(identities[0]);
+      // Process options that were set
+      if (providerIds.length === 0) {
+        Object.keys(identitiesManager.getProviders()).forEach((id) =>
+          providerIds.push(id as ProviderId),
+        );
+      }
```

Comment:
Just confirming that this is desired - i.e. no providers provided, so we perform the search on **all** providers.

Reply:
Yeah thats intended - the `providerId` is optional so if you don't provide one you're asking to search across all providers. So for the case of `identitiesInfoGet` you're searching for the same handle across all providers.

Reply:
Great, resolved.